raft/src/test/java/org/apache/kafka/raft/LeaderStateTest.java

```diff
@@ -789,6 +790,27 @@ public void testGrantVote(boolean isLogUpToDate) {
         assertFalse(state.canGrantVote(ReplicaKey.of(3, Optional.empty()), isLogUpToDate));
     }
 
+    @Test
+    public void testBeginQuorumEpochTimer() {
+        MockTime time = new MockTime();
+        int leader = localId;
+        int follower1 = 1;
+        long epochStartOffset = 10L;
+
+        Set<Integer> voterSet = mkSet(leader, follower1);
+        LeaderState<?> state = newLeaderState(voterSet, epochStartOffset);
+
+        long resetTime = time.milliseconds();
+        state.resetBeginQuorumEpochTimer(resetTime);
```

Comment:
Before reseting the timer you should check that the expiration time is what you expect when becoming leader.